### PR TITLE
[AOT] Add CreateExecutorMetadata analysis pass

### PR DESCRIPTION
**Pull request by mbaret**
Tracking issue: https://github.com/apache/tvm/issues/12548

AOT requires the ExecutorCodegenMetadata object to be populated containing various pieces of information about the compiled module. This commit adds a separate analysis pass to create the metadata + some tests for the new pass.

### CODE REVIEW

**Comment by mbaret** [comment]
cc also @ekalda

File: tests/python/relay/aot/test_aot_create_executor_metadata.py [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+            f.params[2]: workspace_pool_info,
+            f.params[3]: constant_pool_info,
+        },
+        "io_pool_allocations": io_pool_allocations,
+    }
+
+    executor_metadata = CreateExecutorMetadata(mod, "test_mod", executor, 16, 1)
+
+    _check_executor_metadata(executor_metadata, expected_metadata)
+
```
Comment (ekalda) [inline tests/python/relay/aot/test_aot_create_executor_metadata.py:125]: Maybe worth checking that nothing bad happens when pool info has not been provided?

**Review by ekalda — COMMENTED** [review]
Looks good! Didn't even find any typos...

File: tests/python/relay/aot/test_aot_create_executor_metadata.py (hunk 1 above)
Comment (mbaret) [reply tests/python/relay/aot/test_aot_create_executor_metadata.py:125]: Added a test case :)

**Review by lhutton1 — APPROVED** [review]
Thanks @mbaret, LGTM!

Just a nit: a test for multiple input and outputs might be helpful? But happy for it to come in a followup

**Comment by ekalda** [comment]
@tvm-bot rerun

**Comment by lhutton1** [comment]
Thanks @mbaret @ekalda!

**Pull request merged**